src/core/server/logging/README.md

```diff
@@ -264,6 +264,125 @@ The maximum number of files to keep. Once this number is reached, oldest files w
 
 The default value is `7`
 
+### Rewrite Appender
+
+*This appender is currently considered experimental and is not intended
+for public consumption. The API is subject to change at any time.*
+
+Similar to log4j's `RewriteAppender`, this appender serves as a sort of middleware,
+modifying the provided log events before passing them along to another
+appender.
+
+The most common use case for the `RewriteAppender` is when you want to
+filter or censor sensitive data that may be contained in a log entry.
+In fact, with a default configuration, Kibana will automatically redact
+any `authorization`, `cookie`, or `set-cookie` headers when logging http
+requests & responses.
+
+To configure additional rewrite rules, you'll need to specify a `RewritePolicy`.
+
+#### Rewrite Policies
+
+Rewrite policies exist to indicate which parts of a log record can be
+modified within the rewrite appender.
+
+**Meta**
+
+The `meta` rewrite policy can read and modify any data contained in the
+`LogMeta` before passing it along to a destination appender.
+
+```yaml
+logging:
+  appenders:
+    my-rewrite-appender:
+      type: rewrite
+      appenders: [console, file] # name of "destination" appender(s)
+      policy:
+        # ...
+```
+
+Meta policies must specify one of three modes, which indicate which action
+to perform on the configured properties:
+- `add` creates a new property at the provided `path`, skipping properties which already exist.
```

Comment:
Yes exactly, `add` creates a new property only if it doesn't already exist (to avoid unexpectedly overwriting a value).

Conversely, `update` changes a property only if it already exists (to avoid unexpectedly creating a new value).